source/vk/test/UnitVK/source/CreateInstance.cpp

```diff
@@ -85,10 +85,20 @@ TEST_F(CreateInstance, DefaultExtension) {
                                    nullptr, &extensionCount, nullptr));
 
   if (0 < extensionCount) {
+    uint32_t storedExtensionCount;
     std::vector<VkExtensionProperties> extensionProperties(extensionCount);
-    ASSERT_EQ_RESULT(VK_SUCCESS,
-                     vkEnumerateInstanceExtensionProperties(
-                         nullptr, &extensionCount, extensionProperties.data()));
+
+    storedExtensionCount = 0;
+    ASSERT_EQ_RESULT(VK_INCOMPLETE, vkEnumerateInstanceExtensionProperties(
+                                        nullptr, &storedExtensionCount,
+                                        extensionProperties.data()));
+    ASSERT_EQ(0, storedExtensionCount);
+
+    storedExtensionCount = 0xffffffff;
```

Comment:
should this not be no bigger than the size of the vector? (I appreciate it shrinks to the number of extensions, but even so).

Reply:
In non-test code, for writing sensible code, absolutely, this is silly. But it's technically valid, and as a test, this is important to cover one of the cases that failed: on 32-bit platforms, depending on the optimization level, `vk::instance_extensions.begin() + *pPropertyCount` would end up as `vk::instance_extensions.begin() - 1` (ignoring some details) and therefore do the wrong thing because `std::min()` would not consider it greater than `vk::instance_extensions.end()`.

Reply:
could we make the std::vector that big then? If you prefer not to, then I'm fine as is.

Reply:
fair enough, it makes sense -  maybe a comment explaining why 0xffffffff

Reply:
Yep, a comment makes sense, will do.